scripts/setup-new-module

```diff
@@ -0,0 +1,352 @@
+    if [ ! -f "$release_maven" ]; then
+        echo "Warning: $release_maven does not exist. Skipping."
+    else
+        # Update MODULES_TO_SKIP in release-to-maven.yml
+        if grep -q "MODULES_TO_SKIP=" "$release_maven"; then
+            # Extract current value
+            current_modules=$(grep "MODULES_TO_SKIP=" "$release_maven" | cut -d'"' -f2)
+            # Add new module to skip
+            new_modules="$current_modules,$module_name"
+            # Update the file
+            sed -i.bak "s/MODULES_TO_SKIP=\"$current_modules\"/MODULES_TO_SKIP=\"$new_modules\"/" "$release_maven"
+            rm -f "${release_maven}.bak"
+            echo "Updated MODULES_TO_SKIP in $release_maven to include $module_name"
+        else
+            echo "MODULES_TO_SKIP variable not found in $release_maven. Please manually update."
+        fi
+    fi
+    
+    if [ ! -f "$release_javadoc" ]; then
+        echo "Warning: $release_javadoc does not exist. Skipping."
+    else
+        # Update MODULES_TO_SKIP in release-javadoc.yml
+        if grep -q "MODULES_TO_SKIP=" "$release_javadoc"; then
+            # Extract current value
+            current_modules=$(grep "MODULES_TO_SKIP=" "$release_javadoc" | cut -d'"' -f2)
+            # Add new module to skip
+            new_modules="$current_modules,$module_name"
+            # Update the file
+            sed -i.bak "s/MODULES_TO_SKIP=\"$current_modules\"/MODULES_TO_SKIP=\"$new_modules\"/" "$release_javadoc"
+            rm -f "${release_javadoc}.bak"
+            echo "Updated MODULES_TO_SKIP in $release_javadoc to include $module_name"
+        else
+            echo "MODULES_TO_SKIP variable not found in $release_javadoc. Please manually update."
+        fi
+    fi
+}
+
+# Perform updates based on module type
+if [ "$IS_TEST_MODULE" = false ]; then
+    echo "Performing non-test module updates..."
+    
+    # Add to tests-coverage-reporting pom.xml
+    add_dependency_to_pom "$ROOT_DIR/test/tests-coverage-reporting/pom.xml" "$MODULE_NAME"
+    
+    # Add to aws-sdk-java pom.xml
+    add_dependency_to_pom "$ROOT_DIR/aws-sdk-java/pom.xml" "$MODULE_NAME"
+    
+    # Add to architecture-tests pom.xml
+    add_dependency_to_pom "$ROOT_DIR/test/architecture-tests/pom.xml" "$MODULE_NAME"
+    
+    # Add to bom pom.xml
+    add_dependency_to_pom "$ROOT_DIR/bom/pom.xml" "$MODULE_NAME"
+    
+    # Update japicmp plugin config
+    update_japicmp_config "$MODULE_NAME"
+    
+    # Update .brazil.json
+    update_brazil_json "$MODULE_NAME" false
+    
+    # Update root pom.xml
+    update_root_pom "$MODULE_NAME"
+    
+else
+    echo "Performing test module updates..."
+    
+    # Update buildspecs
+    update_buildspecs "$MODULE_NAME"
+    
+    # Update .brazil.json
+    update_brazil_json "$MODULE_NAME" true
+    
+    # Update root pom.xml
+    update_root_pom "$MODULE_NAME"
+fi
+
+echo ""
+echo "Module setup complete! Please review the changes and complete any manual steps mentioned above."
```

Comment:
I'm not sure how much output this produces, but it might be good to collect the required manual steps and provide them all together here so they are less likely to be missed. (feel free to ignore if its never much output).

Reply:
There's actually no manual steps anymore :) Will update it